texar/data/data/multi_aligned_data.py

```diff
@@ -220,35 +217,48 @@ def _get_filter(delimiter, max_seq_length):
                                                hparams_i["max_seq_length"]))
                 else:
                     filters.append(None)
+
+                if _is_text_data(dtype):
+                    self._names.append(
+                        {"text": connect_name(hparams_i["data_name"], "text"),
+                         "text_ids": connect_name(hparams_i["data_name"],
+                                                  "text_ids"),
+                         "length": connect_name(hparams_i["data_name"],
+                                                "length")})
+
+                    text_hparams = MonoTextData.default_hparams()
+                    text_hparams.update({"dataset": hparams_i.todict()})
+
+                    # MonoTextData does not have following keys
+                    for key in ["data_type", "vocab_share_with",
+                                "embedding_init_share_with",
+                                "processing_share_with"]:
+                        del text_hparams["dataset"][key]
+
+                    self._databases.append(
+                        MonoTextData._construct(hparams=text_hparams,
+                                                device=device,
+                                                vocab=self._vocab[idx],
+                                                embedding=self._embedding[idx]))
+                else:
+                    self._names.append({"label": hparams_i["data_name"]})
+                    scalar_hparams = ScalarData.default_hparams()
+                    scalar_hparams.update({"dataset": hparams_i.todict()})
+                    self._databases.append(ScalarData(hparams=scalar_hparams,
```

Comment:
What about the source of this `ScalarData`? Won't this data base load all data and process them internally?
Same applies to `RecordData`.

Reply:
Created a `_construct` method in `ScalarData` and `RecordData` to solve this.